test/functional/services/dashboard/add_panel.ts

```diff
@@ -24,14 +24,18 @@ export class DashboardAddPanelService extends FtrService {
     await this.common.sleep(500);
   }
 
-  async clickCreateNewLink() {
+  async clickCreateNewLink(skipLoadingIndicatorHiddenCheck?: boolean) {
     this.log.debug('DashboardAddPanel.clickAddNewPanelButton');
     await this.retry.try(async () => {
       // prevent query bar auto suggest from blocking button
       await this.browser.pressKeys(this.browser.keys.ESCAPE);
       await this.testSubjects.click('dashboardAddNewPanelButton');
       await this.testSubjects.waitForDeleted('dashboardAddNewPanelButton');
-      await this.header.waitUntilLoadingHasFinished();
+
+      if (!skipLoadingIndicatorHiddenCheck) {
```

Comment:
Why do you want to skip waiting for loading to finish? This may introduce flakiness as without waiting for loading to finish, exists clause below may fail if loading has not finished.

Reply:
This is an issue in serverless tests. 
It throws timeout error waiting for `Find.existsByCssSelector('[data-test-subj="globalLoadingIndicator-hidden"]') with timeout=100000`

Reply:
> Update: @pheyos is looking into the issue why loading indicator not working in serverless.

This is going to be a problem with most functional tests. The right solution will be to solve the problem in `waitUntilLoadingHasFinished` for serverless and not remove its usage on a case by case basis.

Reply:
I've looked into this and found that #158523 moved the global loading indicator for projects, including a different `data-test-subj`. I've reached out to SharedUX folks to see if we can make both loading indicators follow the same naming for `data-test-subj`. If so, we wouldn't have to change anything on the test side. If not, we'd have to adjust things in `waitUntilLoadingHasFinished` as @nreese said. I'll keep you posted.

Reply:
As of now I have `removed the skip flag` and `added retry ` in tests.

Reply:
The feedback I've got so far was to align stateful and serverless versions of the loading indicator `data-test-subj` names, so I've raised a PR with a suggested fix: #163697. Let's see if that finds agreement.

Reply:
My PR was accepted and merged, so we should be able to use `waitUntilLoadingHasFinished` without modifications in serverless. Let me know if you come across any further issues with it.